sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,704 @@
+//! @brief Solana Rust-based BigNumber for bpf-programs
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use std::fmt;
+
+#[repr(transparent)]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+    // Clone,
+    Default,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    AbiExample,
+)]
+
+pub struct BigNumber(u64);
+
+// Syscall interfaces
+#[cfg(target_arch = "bpf")]
+extern "C" {
```

Comment:
Stick these in their representative functions below

Reply:
Fixed in 03af7519e1ca7a29ed2ee25c2cd6b31e0d4e99fd